pkg/restore/restore.go

```diff
@@ -118,10 +121,8 @@ type kubernetesRestorer struct {
 // NewKubernetesRestorer creates a new kubernetesRestorer.
 func NewKubernetesRestorer(
 	restoreClient velerov1client.RestoresGetter,
-	discoveryHelper discovery.Helper,
-	dynamicFactory client.DynamicFactory,
+	client kbclient.Client,
```

Comment:
I would keep the params for the old clients here to keep the diff small but remove the fields and not set them in the struct (afaik like we did in backup)